test/gradient.jl

```diff
@@ -55,7 +55,20 @@ using Test
                            KernelFactors.scharr, KernelFactors.bickley,
                            Kernel.ando3, Kernel.sobel, Kernel.prewitt,
                            Kernel.scharr, Kernel.bickley)
-            gy, gx, gz = @inferred(imgradients(img, kernelfunc, Inner()))
+            # when inference improves, use this instead of following blocks:
+            # gy, gx, gz = @inferred(imgradients(img, kernelfunc, Inner()))
+
+# begin substitute
+            flag = false
+            try
+                res = @inferred(imgradients(img, kernelfunc, Inner()))
+            catch
+                flag = true
+            end
+            @test_broken flag == false
+# end substitute
```

Comment:
Ideally we'd get this and the one below fixed. May not be vital though if it slows down merging any more.